src/Core/System/SystemConfig/SystemConfigService.php

```diff
@@ -227,11 +241,14 @@ public function set(string $key, $value, ?string $salesChannelId = null): void
             return;
         }
 
+        $event = new BeforeSystemConfigChangedEvent($key, $value, $salesChannelId);
```

Comment:
I am not triggering it for deletion by design as it makes little sense there, if it should be a different event, but if you disagree i can add it there too.